.github/workflows/test.yml

```diff
@@ -244,7 +261,12 @@ jobs:
           name: "${{ github.sha }}-ab"
         if: "env.GIT_DIFF != ''"
       - name: Run tests with race detector
-        run: cat xab.txt | xargs go test -mod=readonly -timeout 15m -race -tags='cgo ledger test_ledger_mock'
+        run: cat xab.txt | xargs go test -mod=readonly -json -timeout 15m -race -tags='cgo ledger test_ledger_mock' > xab-race-output.txt
+        if: "env.GIT_DIFF != ''"
+      - uses: actions/upload-artifact@v2
```

Comment:
Again, `if: env.GIT_DIFF` might be missing here.

Reply:
Ditto